include/circt/Dialect/HW/HWMiscOps.td

```diff
@@ -82,3 +82,23 @@ def ParamValueOp : HWOp<"param.value",
   let hasVerifier = 1;
   let hasFolder = true;
 }
+
+def EnumConstantOp : HWOp<"enum.constant", [NoSideEffect, ConstantLike,
```

Comment:
I was initially considering exactly this + making `EnumType` an `HWIntegerType` and it may still be a reasonable move - it would allow us to use all of the other ops in `hw` (`constant`,`array`,...).  
My reasoning for not going that route is:
* We're claiming nothing about the underlying type and encoding (as intended, in this commit). However, in reality, that encoding will always be integer-like, in that it regardlessly have to resolve to a set of bits.
* `comb` operations expect `HWIntegerType`'d values. If arithmetic is allowed on enum values in the SV standard then... i guess it would be fine? but it still feels a bit weird. Maybe there should be a new `isHWBitLikeType` wherein `isHWIntegerType` is a refinement of that constraint?

Let me know what you think - if we can find a way to integrate `EnumType` better into the existing `hw` ops, then I'm all for it!

Reply:
Yeah, I guess the question is how the SV spec treats these things. If it's non-sensical to do things like add two enums, then let's definitely not do that. I was only really considering the ability to re-use `hw.constant`, not the can of worms we'd potentially open if we allow these things to parade around as integers anywhere in the comb dialect.

Reply:
Maybe it's also worth considering that hw/comb shouldn't be held hostage by the possibly bad/bloated design decisions made in the system verilog spec. I think a more reasonable approach would be that, if a user really wants to use an enum value as an integer-like value, there should be an explicit cast operation to perform to facilitate this.

Reply:
+1

Reply:
I'm already now running into issues with wanting to gather enums in `hw` arrays (states are enumerated, those are placed in a `hw.array` and selected as the next state).

How about if we make enums Integer-like for now, and create an issue to remodel the HW constraints into being `isHWScalarLike` (everything that can go into arrays, constants, ...) with a refinement being `isHWIntegerLike` which is referenced by `comb`?

Reply:
Cast from enum type to interger type makes a lot of sense.

Reply:
It may or may not, but I don’t think it’s strictly necessary at this point. @mortbopet please correct me if I’m wrong or it makes something significantly easier.

Reply:
While the integer cast could in theory be considered separate from the encoding considerations (that is, the actual cast is realized by, upon SV emission, enum values are emitted in place of where the cast integer value is used, and then it is up to the tool/SV standard to hopefully do things correctly), I'd prefer this to be deferred to a future commit which tackles all of these things at once - base type, encoding, and integer casting. I see these as separate from the base utility that this commit provides (and which solves our immediate needs).